### ISSUE TITLE
Invalid `toString()` method generated for modifiable class with encodings

### ISSUE THREAD

**Issue by chrisbouchard** [immutables/immutables#486]
I have a class [`TaxInfo`](#user-content-TaxInfo) annotated with `@Value.Modifiable`. The class also has a field `setups` of type `ImmutableTable`, and there is an encoding [`TableEncoding`](#user-content-TableEncoding) for that type. The `ImmutableTaxInfo` class generates fine, but `ModifiableTaxInfo` has an error. The relevant section of that generated file is:

```java
/**
 * Generates a string representation of this {@code TaxInfo}.
 * If uninitialized, some attribute values may appear as question marks.
 * @return A string representation
 */
@Override
public String toString() {
  return MoreObjects.toStringHelper("ModifiableTaxInfo")
      .add("description", descriptionIsSet() ? description() : "?")
      .add("setups", setupsIsSet() ? setups() : "?")
      .add("productSetups", productSetups())
      .toString();
}
```

The second call to `add` tries to call `setupsIsSet`, but that method was not generated. I see that `descriptionIsSet` is generated, but that `productsIsSet` is neither generated nor called. Is there anything I can do in my encoding to either generate this method, or cause it not to be called? I expect the default value to be an empty table, so it should never be unset.

I will just mention that I'm only trying to use modifiable classes so I can use Immutables with JPA/Hibernate. If there is a better way, please let me know.

#### Specifics:
- Immutables 2.3.8
- Java JDK 1.8.0_91
- Eclipse Oxygen (4.7)
- Gradle 3.1 with apt 0.8
- Windows 10

---
#### <a name="TaxInfo"></a> `TaxInfo`:
```java
@JsonDeserialize(builder = ImmutableTaxInfo.Builder.class)
@JsonSerialize(as = ImmutableTaxInfo.class)
@Value.Immutable
@Value.Modifiable
public abstract class TaxInfo {

    public abstract String description();

    public abstract ImmutableTable<PaidAt, DealType, TaxSetup> setups();

    public abstract ImmutableMap<Product, TaxSetup> productSetups();

    /* Other methods ommitted... */

}
```

#### <a name="TableEncoding"></a> `TableEncoding`:
```java
@Encoding
public final class TableEncoding<R, C, V> {

    @Encoding.Impl private final ImmutableTable<R, C, V> value = ImmutableTable.of();

    @Encoding.Expose
    ImmutableTable<R, C, V> get() {
        return value;
    }

    @Override
    public int hashCode() {
        return value.hashCode();
    }

    @Override
    public String toString() {
        return value.toString();
    }

    boolean equals(TableEncoding<R, C, V> other) {
        return value.equals(other.value);
    }

    @Encoding.Of
    static <R, C, V> ImmutableTable<R, C, V> of(Table<? extends R, ? extends C, ? extends V> table) {
        return ImmutableTable.copyOf(table);
    }

    @Encoding.Builder
    static class Builder<R, C, V> {
        private final ImmutableTable.Builder<R, C, V> builder = ImmutableTable.<R, C, V>builder();

        @Encoding.Copy
        @Encoding.Init
        @Encoding.Naming(standard = INIT)
        void init(Table<? extends R, ? extends C, ? extends V> table) {
            putAll(table);
        }

        @Encoding.Init
        @Encoding.Naming(standard = PUT)
        void put(R row, C column, V value) {
            builder.put(row, column, value);
        }

        @Encoding.Init
        @Encoding.Naming(standard = PUT)
        void putCell(Table.Cell<? extends R, ? extends C, ? extends V> cell) {
            builder.put(cell);
        }

        @Encoding.Init
        @Encoding.Naming(standard = PUT_ALL)
        void putAll(Table<? extends R, ? extends C, ? extends V> table) {
            builder.putAll(table);
        }

        @Encoding.Build
        ImmutableTable<R, C, V> build() {
            return builder.build();
        }
    }

}
```

**Comment by chrisbouchard** [comment]
Also, `ModifiableTaxInfo.setSetups` has the signature
```java
public ModifiableTaxInfo setSetups(ImmutableTable<PaidAt, DealType, TaxSetup> setups)
```
rather than
```java
public ModifiableTaxInfo setSetups(Table<? extends PaidAt, ? extends DealType, ? extends TaxSetup> setups)
```
which I would expect based on my `@Encoding.Of` method signature.

**Comment by elucash** [comment]
@chrisbouchard thank you for reporting this! Currently, encodings are not integrated into modifiable object generation. The tough question is if they will ever be. Need to investigate the gap as some features of encoding are clearly at odds with how modifiable implementations are structured. There are chances we can "fix" more general cases, leaving less used parts as broken.

**Comment by chrisbouchard** [comment]
While it would be sad if the two features weren't integrated, at the very least the presence of an encoding shouldn't break the modifiable class. I would expect it to just generate as if the encoding didn't exist.

**Comment by elucash** [comment]
totally agree. Need to see how far we are from supporting it, if too far, then we would be better gracefully generate simple/"uncustomized" code.

**Comment by chrisbouchard** [comment]
I can definitely see how it would be difficult to generate a modifiable class using the existing encoding scheme. Even in my example above, the class stores an `ImmutableTable`, but the builder stores an `ImmutableTable.Builder`. I don't think it would be too onerous to have a separate class marked `@Encoding.Modifiable` like you do for builders. Even if some code winds up getting duplicated, it only has to be written once.

For my use case, I'd be ok with not getting any of the builder methods, but I would really like to see the signature of `@Encoding.Of` used for the `set*` method.

**Comment by elucash** [comment]
support for modifiables encodings is not planned yet, but we've fixed the `toString` generation. released in 2.4.4. thank you!
